src/main/java/seedu/contax/ui/onboarding/Overlay.java

```diff
@@ -64,54 +63,110 @@ public void showOnly(DoubleProperty boundX, DoubleProperty boundY,
-        topOverlay.minHeightProperty().bind(boundY);
-        topOverlay.minWidthProperty().bind(parentWidth);
-        topOverlay.maxHeightProperty().bind(boundY);
-        topOverlay.maxWidthProperty().bind(parentWidth);
-
-
-        bottomOverlay.layoutXProperty().bind(boundX);
-        bottomOverlay.layoutYProperty().bind(boundY.add(height));
-        bottomOverlay.minHeightProperty().bind(parentHeight.subtract(boundY.add(height)));
-        bottomOverlay.minWidthProperty().bind(parentWidth);
-        bottomOverlay.maxHeightProperty().bind(parentHeight.subtract(boundY.add(height)));
-        bottomOverlay.maxWidthProperty().bind(parentWidth);
+    /**
+     * Sets height of given overlay with the given ReadOnlyDoubleProperty.
+     * @param overlay overlay to set.
+     * @param property property to bind with.
+     */
+    private void setHeight(Pane overlay, ReadOnlyDoubleProperty property) {
+        overlay.minHeightProperty().bind(property);
+        overlay.maxHeightProperty().bind(property);
+    }
+
+    /**
+     * Sets height of given overlay with the given DoubleBinding.
+     * @param overlay overlay to set.
+     * @param property property to bind with.
+     */
+    private void setHeight(Pane overlay, DoubleBinding property) {
+        overlay.minHeightProperty().bind(property);
+        overlay.maxHeightProperty().bind(property);
+    }
+
+    /**
+     * Sets the given overlay with the given x and y ReadOnlyDoubleProperty.
+     * @param overlay overlay to set.
+     * @param x property to bind x with
+     * @param y property to bind y with
+     */
+    private void setLayout(Pane overlay, ReadOnlyDoubleProperty x, ReadOnlyDoubleProperty y) {
+        overlay.layoutXProperty().bind(x);
+        overlay.layoutYProperty().bind(y);
+    }
+
+    /**
+     * Sets the given overlay with the given x ReadOnlyDoublePropery and y DoubleBinding.
+     * @param overlay overlay to set.
+     * @param x property to bind x with
+     * @param y property to bind y with
+     */
+    private void setLayout(Pane overlay, ReadOnlyDoubleProperty x, DoubleBinding y) {
+        overlay.layoutXProperty().bind(x);
+        overlay.layoutYProperty().bind(y);
     }
 
     /**
-     * Show all overlays
+     * Shows the given overlay
+     * @param overlay
      */
-    private void showAll() {
-        topOverlay.setVisible(true);
-        bottomOverlay.setVisible(true);
+    private void showOverlay(Pane overlay) {
+        overlay.setVisible(true);
+        overlay.setStyle("-fx-background-color: rgb(0, 0, 0, 0.5)");
     }
 
     /**
-     * Hide all overlays
+     * Hides the given overlay
+     * @param overlay
      */
-    private void hideAll() {
-        topOverlay.setVisible(false);
-        bottomOverlay.setVisible(false);
+    private void hideOverlay(Pane overlay) {
+        overlay.setVisible(false);
+        overlay.setStyle("-fx-background-color: rgb(0, 0, 0, 0)");
```

Comment:
Similarly, check if `rgb` should be `rgba`.

Reply:
Updated